docs/book/src/providers.md

```diff
@@ -37,3 +37,12 @@ To implement a secrets-store-csi-driver provider, you can develop a new provider
 - Provider mounts `<kubelet root dir>/pods` (default: [`/var/lib/kubelet/pods`](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/v0.0.14/deploy/secrets-store-csi-driver.yaml#L86-L87)) with [`HostToContainer` mount propagation](https://kubernetes-csi.github.io/docs/deploying.html#driver-volume-mounts) to be able to write the external secrets store content to the volume target path
 
 See [design doc](https://docs.google.com/document/d/10-RHUJGM0oMN88AZNxjOmGz0NsWAvOYrWUEV-FbLWyw/edit?usp=sharing) for more details.
+
+## Features supported by current providers:
+
+| Features \ Providers               | Azure | GCP   | AWS   | Vault |
+| ---------------------------------- | ----- | ----- | ----- | ----- |
+| Sync as Kubernetes secret          | Yes   | Yes   | Yes   | Yes   |
+| Rotation                           | Yes   | Yes   | Yes   | No    |
```

Comment:
What should the provider do to be considered to support rotation? The experience could certainly do with some polish because we don't do any caching, but if someone turns the feature on in the driver flags, I would expect it to successfully fetch credentials every time the Vault provider is asked for them.

Reply:
@tomhjp We mainly look at e2e tests to see if feature is supported. Does Vault provider has rotation tests?

Reply:
There aren't any currently. My view was that it's a driver feature so there's no need to add tests in the provider. From the provider's point of view, is there anything different between one mount request that gets triggered by pod creation and another that gets triggered by the auto rotation feature?

Reply:
The mount request is the same in both cases. But I think it'll be a good idea to validate e2e scenario and make sure rotation mount request actually able to rotate secret content in the pod. WDYT?

Reply:
We can add a test case if you like. FWIW I've tested it locally and secret rotation does work as expected.

Reply:
I think we should add a test. But I am happy to hear others thoughts as well.

Reply:
I made a small test case here: https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/758. I think it probably makes sense to require tests for features (though for GCP I only do the test in the GCP repo for permission issues).

Reply:
Thanks the PR @tam7t